File: bdp-govern/atlas-versions/atlas-3.0.0-SNAPSHOT/atlas-examples/sample-app/src/main/java/org/apache/atlas/examples/sampleapp/GlossaryExample.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.atlas.examples.sampleapp;

import org.apache.atlas.AtlasClientV2;
import org.apache.atlas.model.glossary.AtlasGlossary;
import org.apache.atlas.model.glossary.AtlasGlossary.AtlasGlossaryExtInfo;
import org.apache.atlas.model.glossary.AtlasGlossaryCategory;
import org.apache.atlas.model.glossary.AtlasGlossaryTerm;
import org.apache.atlas.model.glossary.relations.AtlasGlossaryHeader;

public class GlossaryExample {
    private static final String GLOSSARY_NAME = "EmployeeCountry";

    private final AtlasClientV2         client;
    private       AtlasGlossary         empGlossary;
    private       AtlasGlossaryTerm     empSalaryTerm;
    private       AtlasGlossaryCategory empCompanyCategory;

    GlossaryExample(AtlasClientV2 client) {
        this.client = client;
    }

    public void createGlossary() throws Exception {
        AtlasGlossary glossary = new AtlasGlossary();

        glossary.setName(GLOSSARY_NAME);
        glossary.setLanguage("English");
        glossary.setShortDescription("This is a test Glossary");

        empGlossary = client.createGlossary(glossary);
    }

    public void getGlossaryDetail() throws Exception {
        AtlasGlossaryExtInfo extInfo = client.getGlossaryExtInfo(empGlossary.getGuid());

        assert (extInfo != null);

        SampleApp.log("Glossary extended info: " + extInfo.getGuid() + "; name: " + extInfo.getName() + "; language: " + extInfo.getLanguage());
    }

    public void createGlossaryTerm() throws Exception {
        if (empSalaryTerm != null) {
            SampleApp.log("EmpSalaryTerm: term already exists");
            return;
        }

        AtlasGlossaryHeader glossary = new AtlasGlossaryHeader();
        AtlasGlossaryTerm   term     = new AtlasGlossaryTerm();

        glossary.setGlossaryGuid(empGlossary.getGuid());
        glossary.setDisplayText(empGlossary.getName());

        term.setAnchor(glossary);
        term.setName("EmpSalaryTerm");

        empSalaryTerm = client.createGlossaryTerm(term);

        if (empSalaryTerm != null) {
            SampleApp.log("Created term for Employee Salary: " + empSalaryTerm);
        }
    }

    public void createGlossaryCategory() throws Exception {
        if (empCompanyCategory != null) {
            SampleApp.log("EmpSalaryCategory: category already exists");
            return;
        }

        AtlasGlossaryHeader   glossary  = new AtlasGlossaryHeader();
        AtlasGlossaryCategory category = new AtlasGlossaryCategory();

        glossary.setGlossaryGuid(empGlossary.getGuid());
        glossary.setDisplayText(empGlossary.getName());

        category.setAnchor(glossary);
        category.setName("EmpSalaryCategory");

        empCompanyCategory = client.createGlossaryCategory(category);

        if (empCompanyCategory != null) {
            SampleApp.log("Created Category for Employee Category :- " + empCompanyCategory);
        }
    }

    public void deleteGlossary() throws Exception {
        if (empGlossary != null) {
            client.deleteGlossaryByGuid(empGlossary.getGuid());

            SampleApp.log("empGlossary is not present. Skipping the delete operation.");
        }

        empGlossary        = null;
        empSalaryTerm      = null;
        empCompanyCategory = null;
    }
}

File: govern/data-meta/atlas/atlas-examples/sample-app/src/main/python/glossary_example.py
#!/usr/bin/env/python

#
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

import logging

from apache_atlas.model.glossary import AtlasGlossary, AtlasGlossaryCategory, AtlasGlossaryTerm, AtlasGlossaryHeader

LOG = logging.getLogger('glossary-example')


class GlossaryExample:
    glossaryName = "EmployeeCountry"

    def __init__(self, client):
        self.client               = client
        self.emp_glossary         = None
        self.emp_salary_term      = None
        self.emp_company_category = None

    def create_glossary(self):
        glossary          = AtlasGlossary({ 'name': GlossaryExample.glossaryName, 'shortDescription': 'This is a test Glossary' })
        self.emp_glossary = self.client.glossary.create_glossary(glossary)

        LOG.info("Created glossary with name: %s and guid: %s", self.emp_glossary.name, self.emp_glossary.guid)

        return self.emp_glossary

    def get_glossary_detail(self):
        ext_info = self.client.glossary.get_glossary_ext_info(self.emp_glossary.guid)

        if ext_info:
            LOG.info("Glossary extended info: %s; name: %s; language: %s", ext_info.guid, ext_info.name, ext_info.language)

    def create_glossary_term(self):
        header = AtlasGlossaryHeader({ 'glossaryGuid': self.emp_glossary.guid, 'displayText': self.emp_glossary.name })
        term   = AtlasGlossaryTerm({ 'name': 'EmpSalaryTerm', 'anchor': header })

        self.emp_salary_term = self.client.glossary.create_glossary_term(term)

        if self.emp_salary_term:
            LOG.info("Created Term for Employee Salary: %s with guid: %s", self.emp_salary_term.name, self.emp_salary_term.guid)

    def create_glossary_category(self):
        header   = AtlasGlossaryHeader({ 'glossaryGuid': self.emp_glossary.guid, 'displayText': self.emp_glossary.name })
        category = AtlasGlossaryCategory({ 'name': 'EmpSalaryCategory', 'anchor': header })

        self.emp_company_category = self.client.glossary.create_glossary_category(category)

        if self.emp_company_category:
            LOG.info("Created Category for Employee Category: %s with guid: %s", self.emp_company_category.name, self.emp_company_category.guid)

    def delete_glossary(self):
        if not self.emp_glossary:
            LOG.info("empGlossary is not present. Skipping the delete operation.")

        self.client.glossary.delete_glossary_by_guid(self.emp_glossary.guid)

        LOG.info("Delete is complete for Glossary!")